src/trix/models/composition.coffee

```diff
@@ -211,7 +219,13 @@ class Trix.Composition extends Trix.BasicObject
     @currentAttributes[attributeName]?
 
   toggleCurrentAttribute: (attributeName) ->
-    if value = not @currentAttributes[attributeName]
+    if ("header" in @getBlock()?.getAttributes() and 
+    attributeName in @getBlock()?.getAttributes() and
+    attributeName != "header")
```

Comment:
This is difficult to read. Can you summarize what this condition checks and what happens when true?